kadi/commands/observations.py

```diff
@@ -0,0 +1,395 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+import os
+from collections import defaultdict
+import logging
+from pathlib import Path
+
+import numpy as np
+from cxotime import CxoTime
+import astropy.units as u
+from astropy.table import Table
+
+# kadi.commands.commands_v2 is also a dependency but encapsulated in the
+# functions to reduce top-level imports for v1-compatibility
+
+logger = logging.getLogger(__name__)
+
+
+__all__ = ['get_starcats', 'get_observations']
+
+AGASC_FILE = Path(os.environ['SKA'], 'data', 'agasc', 'proseco_agasc_1p7.h5')
+
+TYPE_MAP = ['ACQ', 'GUI', 'BOT', 'FID', 'MON']
+IMGSZ_MAP = ['4x4', '6x6', '8x8']
+PAR_MAPS = [
+    ('imnum', 'slot', int),
+    ('type', 'type', lambda n: TYPE_MAP[n]),
+    ('imgsz', 'sz', lambda n: IMGSZ_MAP[n]),
+    ('maxmag', 'maxmag', float),
+    ('yang', 'yang', lambda x: np.degrees(x) * 3600),
+    ('zang', 'zang', lambda x: np.degrees(x) * 3600),
+    ('dimdts', 'dim', int),
+    ('restrk', 'res', int),
+]
+
+# Cache of observations by scenario
+OBSERVATIONS = {}
+
+# Cache of important columns in proseco_agasc_1p7.h5
+STARS_AGASC = None
+
+
+def set_detector_and_sim_offset(aca, simpos):
+    """Get detector from SIM position.
+
+    Finds the detector with nominal SIM position closest to ``simpos``.
```

Comment:
From V&V I seem to recall exceptions to this (ACIS-I vs S) but I suppose it may not matter for this application.

Reply:
Yes, doesn't matter here. Undercover observations will also be incorrectly labeled, but that's OK.